dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -260,9 +257,9 @@ private static bool IsAsyncMethod(MethodInfo method)
     /// <summary>
     /// Gets a delegate for handling the marshaling of a parameter.
     /// </summary>
-    private static (Func<Kernel, ContextVariables, CancellationToken, object?>, KernelParameterMetadata?) GetParameterMarshalerDelegate(
+    private static (Func<Kernel, KernelFunctionArguments, CancellationToken, object?>, KernelParameterMetadata?) GetParameterMarshalerDelegate(
```

Comment:
Mentioned above also, should we support `Task<FunctionResult>` and `FunctionResult` as supported return types?

I think we supported `SKContext` previously.

Reply:
I don't see any reason why we shouldn't. It has already been implemented as part of another comment.